rest/blip_sync.go

```diff
@@ -877,17 +876,25 @@ func (bh *blipHandler) handleRev(rq *blip.Message) error {
 		return base.HTTPErrorf(http.StatusBadRequest, "Missing docID or revID")
 	}
 
-	newDoc := &db.Document{
-		ID:    docID,
-		RevID: revID,
-	}
-	newDoc.UpdateBodyBytes(bodyBytes)
+	hasAttachments := bytes.Contains(bodyBytes, []byte(db.BodyAttachments))
+	hasExpiry := bytes.Contains(bodyBytes, []byte(db.BodyExpiry))
+
+	var body db.Body
+	var newDoc *db.IncomingDocument
+
+	newDoc = db.NewIncomingDocument(docID, bodyBytes)
```

Comment:
Avoid the var declaration above if we don't need it

```suggestion
	newDoc := db.NewIncomingDocument(docID, bodyBytes)
```